ethstorage/miner/l1_mining_api.go

```diff
@@ -350,24 +354,23 @@ func checkGasPrice(
 		txCostCap := new(big.Int).Sub(costCap, extraCost)
 		lg.Debug("Tx cost cap", "txCostCap", txCostCap)
 		profitableGasFeeCap := new(big.Int).Div(txCostCap, new(big.Int).SetUint64(estimatedGas))
-		lg.Info("Minimum profitable gas fee cap", "gasFeeCap", profitableGasFeeCap)
-		if gasFeeCap.Cmp(profitableGasFeeCap) == 1 {
-			profit := new(big.Int).Sub(reward, new(big.Int).Mul(new(big.Int).SetUint64(estimatedGas), gasFeeCap))
+
+		// gasPrice = baseFee + tip would be the cheapest gas price and most likely used for tx inclusion
+		gasPrice := new(big.Int).Sub(gasFeeCap, baseFee)
+		lg.Info("Comparing gas price and profitable gas fee cap", "gasPrice(baseFee+tip)", gasPrice, "profitableGasFeeCap", profitableGasFeeCap)
+		// Drop the tx if baseFee + tip is already higher than the profitable gas fee cap
+		if gasPrice.Cmp(profitableGasFeeCap) == 1 {
+			profit := new(big.Int).Sub(reward, new(big.Int).Mul(new(big.Int).SetUint64(estimatedGas), gasPrice))
 			profit = new(big.Int).Sub(profit, extraCost)
 			droppedMsg := fmt.Sprintf("not enough profit (in ether):\r\n reward %s - gas cost %s - extra cost %s = profit %s < min profit %s",
-				fmtEth(reward), fmtEth(new(big.Int).Mul(new(big.Int).SetUint64(estimatedGas), gasFeeCap)), fmtEth(extraCost), fmtEth(profit), fmtEth(minProfit))
+				fmtEth(reward), fmtEth(new(big.Int).Mul(new(big.Int).SetUint64(estimatedGas), gasPrice)), fmtEth(extraCost), fmtEth(profit), fmtEth(minProfit))
 			return nil, errDropped{reason: droppedMsg}
 		}
+		// Cap the gas fee to be profitable
 		if !useConfig {
 			gasFeeCapChecked = profitableGasFeeCap
 			lg.Info("Using profitable gas fee cap", "gasFeeCap", gasFeeCapChecked)
 		}
-	} else {
```

Comment:
Why do we delete the `else` case?

Reply:
Because gasFeeCap is already baseFee*2 + tip